sdformat_mjcf/src/sdformat_mjcf/mjcf_to_sdformat/converters/world.py

```diff
@@ -27,6 +27,75 @@
 import sdformat as sdf
 
 
+def _is_floating_body(body):
+    freejoint = body.get_children("freejoint")
+    if freejoint:
+        return True
+    all_child_joints = body.get_children("joint")
+    for joint in all_child_joints:
+        if joint.type == "free":
+            return True
+    return False
+
+
+def _add_body_to_model(body, physics, model, modifiers):
+    try:
+        parent_name = body.parent.name
+    except AttributeError:
+        parent_name = "world"
+
+    link = mjcf_body_to_sdf(body,
+                            physics,
+                            body_parent_name=parent_name,
+                            modifiers=modifiers)
+    for camera in body.camera:
+        sensor = mjcf_camera_sensor_to_sdf(camera)
+        if sensor is not None:
+            link.add_sensor(sensor)
+
+    serial_sdf_joints_created = []
+    serial_link_created = [link]
+    small_mass_matrix = MassMatrix3d()
+    small_mass_matrix.set_from_sphere(1e-6, 1)
+    small_inertial = Inertiald()
+    small_inertial.set_mass_matrix(small_mass_matrix)
+
+    for joint_num, joint in enumerate(body.joint):
+        if modifiers is not None:
+            modifiers.apply_modifiers_to_element(joint)
+
+        if joint_num < len(body.joint) - 1:
+            child_name = su.find_unique_name(body, "body",
+                                             f"{body.name}_dummy_{joint_num}")
+            joint_sdf = mjcf_joint_to_sdf(joint, parent_name, child_name)
+            serial_sdf_joints_created.append(joint_sdf)
+            link = sdf.Link()
+            link.set_name(child_name)
+            link.set_pose_relative_to(body.name)
+            link.set_inertial(small_inertial)
+            serial_link_created.append(link)
+            parent_name = child_name
+        else:
+            joint_sdf = mjcf_joint_to_sdf(joint, parent_name, body.name)
+            serial_sdf_joints_created.append(joint_sdf)
+
+        if joint_sdf is not None:
+            model.add_joint(joint_sdf)
+
+    if len(body.joint) == 0 and body.freejoint is None:
```

Comment:
replace `body.freejoint is None` with `_is_floating_body` ?

Reply:
Thanks! https://github.com/gazebosim/gz-mujoco/pull/87/commits/e60b1c741dcf8113200e4ac00073635431043514